src/platform/Darwin/DnssdInterfaceInfos.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "DnssdInterfaceInfos.h"
+#include "DnssdImpl.h"
+
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+#include <net/ethernet.h>
+#include <net/if_dl.h>
+#include <netdb.h>
+
+constexpr DNSServiceFlags kRegisterRecordFlags = kDNSServiceFlagsShared;
+
+namespace chip {
+namespace Dnssd {
+
+namespace {
+
+static void OnRegisterRecord(DNSServiceRef sdRef, DNSRecordRef recordRef, DNSServiceFlags flags, DNSServiceErrorType err,
+                             void * context)
+{
+    ChipLogError(Discovery, "Mdns: %s flags: %d", __func__, flags);
+
+    auto sdCtx = reinterpret_cast<RegisterRecordContext *>(context);
+    VerifyOrReturn(kDNSServiceErr_NoError == err, sdCtx->Finalize(err));
```

Comment:
So if I understand correctly, if any RegisterRecord bit fails we fail the entire Register, even if RegisterRecord succeeded on other interfaces?  Is this the behavior we want?